### ISSUE TITLE
App changes made using App studio doesn't appear on other users MS teams client

### ISSUE THREAD

**Issue by rjgmail88** [OfficeDev/BotBuilder-MicrosoftTeams#163]
I followed this [document](https://docs.microsoft.com/en-us/microsoftteams/platform/get-started/get-started-app-studio) to register a Ms teams App using App studio for my exiting BOT. 

Then added a Menu items as mentioner [here](https://docs.microsoft.com/en-us/microsoftteams/platform/concepts/bots/bots-menus) 

Now everything seems to be on my computer or I should say MS Teams client that I am using. Like I am able to see the Menu items and able to interact with the new App.

When I ask other users , who are using same Bot in their personal scope / 1:1 chat they do not see any updates or Menu items in this case. 

Note: I just clicked Install Button from following options after making changes. 

Does that mean it will install to only my Teams App and doesn't actually make changes to Existing BOT which is being used my many people in their personal chat ? 

![image](https://user-images.githubusercontent.com/32375467/47808481-78c01380-dd0c-11e8-99a9-32225103a26e.png)

Here is what it looks like in App studio where scope is personal for **Commands**. 

![image](https://user-images.githubusercontent.com/32375467/47808681-e0765e80-dd0c-11e8-8e05-3f43e5825340.png)

**Comment by Wajeed-msft** [comment]
@rjgmail88 - bot menus are part of manifest, you need to install the bot again with updated manifest file to see bot menu. Other users will see the bot menu only after re-installing updated manifest file.

**Comment by rjgmail88** [comment]
Hi @Wajeed-msft , But, other users are not able to see anything ? like no Menu items for them. Is that because I clicked Install from my App studio ? and it install it only for me ?

**Comment by Wajeed-msft** [comment]
@rjgmail88 - Yes, you are correct. Personal scope app needs to be installed by individual. For team scope, installing updated package in team will reflect the changes for all the team members. Please download the manifest and ask another user to install in his/her personal scope to see bot menu.

**Comment by rjgmail88** [comment]
@Wajeed-msft , I am confused between definition of scope here. As per my knowledge there are 2 scope personal and team , and those represents how BOT will be added/accessible in Ms teams. 
ex:- I have added a bot to ms teams by using deep link:-  [https://teams.microsoft.com/l/chat/0/0?users=28:{BotId}](https://teams.microsoft.com/l/chat/0/0?users=28:{BotId})

that is called Bot in my personal scope or 1:1 chat. 

Now if I would like to use it for scopes like team, groupchat I would have to create an App using studio which will involve manifest etc. Then users can add it to respective teams channel after publishing to store. 

However,  when you mentioned about personal scope app needs to be installed by individual. What do they need to do ? I was able to do it because I am a developer so I use app studio and I clicked on Install button. 

We need to roll this to 25K plus user from our organization so I am trying to find best way to roll this out.

Any thought on using this [https://docs.microsoft.com/en-us/microsoftteams/platform/get-started/get-started-app-studio](https://docs.microsoft.com/en-us/microsoftteams/platform/get-started/get-started-app-studio)

**Comment by Wajeed-msft** [comment]
You can use the Microsoft Teams Tenant Apps Catalog to test and distribute line-of-business applications to your organization. Please have a look at documentation on how to [Publish apps to the Microsoft Teams Tenant Apps Catalog](https://docs.microsoft.com/en-us/microsoftteams/tenant-apps-catalog-teams).

**Comment by rjgmail88** [comment]
@Wajeed-msft , I followed above document  to distribute an App by using Apps Catalog to test.  Here are the steps:

1. Used **appV1.0.zip** to upload & publish using Tenant  catalog

2. User searched for appV1 in store, and downloaded it to his Ms teams web app (personal scope )

3. Developer sent new version of App to tenant admin **appV2.0.zip** that has minor changes on the functionality.  

4. Tenant admin did reinstalled it. 

5.  User from step 2 don't see any changes reflected. 

Is this expected ? 

@billbliss it will be nice to have your thoughts on this as well.

**Comment by Wajeed-msft** [comment]
@rjgmail88 - Currently, only way to get the updated app is by restarting desktop client or for web browser reloading the browser tab.
